src/algebra/quaternion.lean

```diff
@@ -153,14 +153,48 @@ rfl
      a₁ * b₃ + c₁ * a₂ * b₄ + a₃ * b₁ - c₁ *  a₄ * b₂,
      a₁ * b₄ + a₂ * b₃ - a₃ * b₂ + a₄ * b₁⟩ := rfl
 
+section
+variables [has_smul S R] [has_smul T R] (s : S)
+
+/-
+The `ring R` argument is not used, but it's also much stronger than the other definitions in this
+file need; for instance `quaternion_algebra.has_zero` only really needs `has_zero R`. For
+simplicity we just keep things consistent.
```

Comment:
This means the `ring R` won't be solvable by unification, which will result in a useless typeclass search. I guess that's fine, but it certainly is a bit of a red flag.